modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java

```diff
@@ -425,6 +421,18 @@ private CompletableFuture<AsyncSqlCursor<List<Object>>> querySingle0(
                 .thenCompose(sqlNode -> {
                     boolean rwOp = dataModificationOp(sqlNode);
 
+                    boolean implicitTxRequired = outerTx == null;
+
+                    tx.set(implicitTxRequired ? txManager.begin(!rwOp) : outerTx);
+
+                    SchemaPlus schema = sqlSchemaManager.schema(schemaName);
+
+                    if (schema == null) {
+                        return CompletableFuture.failedFuture(
+                                new IgniteInternalException(SCHEMA_NOT_FOUND_ERR,
```

Comment:
Let's introduce a top-level class SchemaNotFoundException in public API.

Reply:
done